lib/spack/spack/cmd/__init__.py

```diff
@@ -80,46 +84,54 @@ def remove_options(parser, *options):
                 break
 
 
-def get_module(cmd_name):
-    """Imports the module for a particular command name and returns it.
+def get_module_from(cmd_name, namespace):
+    """Imports the module for a particular command from the specified namespace.
 
     Args:
         cmd_name (str): name of the command for which to get a module
             (contains ``-``, not ``_``).
+        namespace (str): namespace for command.
     """
     pname = python_name(cmd_name)
-
-    try:
-        # Try to import the command from the built-in directory
-        module_name = "%s.%s" % (__name__, pname)
-        module = __import__(module_name,
-                            fromlist=[pname, SETUP_PARSER, DESCRIPTION],
-                            level=0)
-        tty.debug('Imported {0} from built-in commands'.format(pname))
-    except ImportError:
-        module = spack.extensions.get_module(cmd_name)
-        if not module:
-            raise
+    module_name = '{0}.cmd.{1}'.format(namespace, pname)
+    module = __import__(module_name,
+                        fromlist=[pname, SETUP_PARSER, DESCRIPTION],
+                        level=0)
+    tty.debug('Imported command {0} as {1}'.format(cmd_name, module_name))
 
     attr_setdefault(module, SETUP_PARSER, lambda *args: None)  # null-op
     attr_setdefault(module, DESCRIPTION, "")
 
     if not hasattr(module, pname):
         tty.die("Command module %s (%s) must define function '%s'." %
                 (module.__name__, module.__file__, pname))
+    return module
+
+
+def get_module(cmd_name):
+    """Imports the module for a particular Spack or extension top-level
+    command name and returns it.
 
+    Args:
+        cmd_name (str): name of the command for which to get a module
+            (contains ``-``, not ``_``).
+    """
+    try:
+        module = get_module_from(cmd_name, 'spack')
```

Comment:
Not entirely sure what you mean by _this_  and _it_ here, sorry. `get_module()` is used in multiple places in core, but `get_module_from()` is called only once.

Reply:
When I read through this I was curious why `get_module_from(name, namespace)` existed since it was only called once and with one argument. The important thing is making it clear to other developers why the function should remain. A test would prevent anyone from removing it. You could also update the docs (https://spack.readthedocs.io/en/latest/extensions.html) with an explanation. You could also expand on the docstring in `get_module_from`:

> Invoke this from command implementations in order to find sub-commands.

I think more detail and an example would be useful